storage/src/main/java/tech/pegasys/teku/storage/store/StoreBuilder.java

```diff
@@ -54,40 +51,49 @@ public static StoreBuilder create() {
   public static UpdatableStore buildForkChoiceStore(
       final MetricsSystem metricsSystem,
       final BlockProvider blockProvider,
-      final BeaconState anchorState) {
-    return forkChoiceStoreBuilder(metricsSystem, blockProvider, anchorState).build();
+      final BeaconState genesisState) {
```

Comment:
`anchorState` is the more correct term (and possibly we should update a lot of places that refer to Genesis).  We might be starting from a state that isn't slot 0.

Reply:
When we process the genesis state, we construct an empty block from the state and apply an empty signature - which I don't think we want to do for non-genesis anchors.  But, I went ahead and changed `GenesisEvent` to `AnchorPoint` so that we have a more generic object to work with, and changed this builder to just work with `AnchorPoint`'s.